### PR TITLE
Remove pandas dependency

### LINKED ISSUE #82: [FEAT]: Remove Pandas dependency
### Description

Since pandas is widely used and often has other dependency restrictions, e.g. on Google Colab, we would encounter

```
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
cudf-cu12 24.4.1 requires pandas<2.2.2dev0,>=2.0, but you have pandas 2.2.2 which is incompatible.
google-colab 1.0.0 requires pandas==2.0.3, but you have pandas 2.2.2 which is incompatible.
```

which doesn't prevent us from install `sotopia==0.0.7`, but it is still annoying. The only use of pandas in our package is converting a dict of list into a csv file, which could be done with various other, even standard libraries. 

Related to the plan mentioned in #80. 

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by ProKil**
Closes #82 

## 📑 Description

Since `pandas` was only used in the serialization, we remove it as a dependency for the core package, but still keep it optional for examples. 

## ✅ Checks

- [x] My pull request adheres to the code style of this project
- [ ] My code requires changes to the documentation
- [ ] I have updated the documentation as required
- [x] All the tests have passed
- [x] Branch name follows `type/descript` (e.g. `feature/add-llm-agents`)
- [x] Ready for code review

### CODE REVIEW

File: sotopia/database/serialization.py [hunk 1]
```diff
@@ -175,6 +176,25 @@ def get_social_interactions_from_episode(
     return "\n\n".join(overall_social_interaction)
 
 
+def _serialize_data_to_csv(data: dict[str, list[Any]], csv_file_path: str) -> None:
```
Comment (XuhuiZhou) [inline sotopia/database/serialization.py:179]: so when outputting things to csv, we should be extremely careful about situations where there are "," in the string of the entry. I don't see that being handled in this function. Pandas automatically handle that so maybe check what they did there?

Another option is to get rid of the csv output and switch everything to jsonl

**Review by XuhuiZhou — CHANGES_REQUESTED** [review]

File: sotopia/database/serialization.py (hunk 1 above)
Comment (ProKil) [reply sotopia/database/serialization.py:179]: This was automatically handled by the builtin csv writer.

Comment (XuhuiZhou) [reply sotopia/database/serialization.py:179]: Right, I could be wrong, but there are many previous cases that I remembered csv.writer could get things wrong or make the file hard to read. 

That's primarily why I switched to pandas

Comment (XuhuiZhou) [reply sotopia/database/serialization.py:179]: have you tested it to see whether you can load the CSV data back? and how?

Comment (ProKil) [reply sotopia/database/serialization.py:179]: Yes, verified in 9729ec4

Comment (XuhuiZhou) [reply sotopia/database/serialization.py:179]: Nice!! Thanks!

**Review by XuhuiZhou — APPROVED** [review]

**Pull request merged**